src/kernels/raw/finder/contributedLocalKernelSpecFinder.node.ts

```diff
@@ -75,28 +75,37 @@ export class ContributedLocalKernelSpecFinder
     activate() {
         this.promiseMonitor.onStateChange(() => {
             this.status =
-                this.promiseMonitor.isComplete && this.interpreters.status === 'idle' ? 'idle' : 'discovering';
+                this.promiseMonitor.isComplete &&
+                this.interpreters.status === 'idle' &&
+                this.nonPythonKernelFinder.status === 'idle' &&
+                this.pythonKernelFinder.status === 'idle'
+                    ? 'idle'
+                    : 'discovering';
         });
-        let deferred: Deferred<void> | undefined = undefined;
-        if (this.interpreters.status === 'refreshing') {
-            deferred = createDeferred<void>();
-            this.promiseMonitor.push(deferred.promise);
-        }
-        this.interpreters.onDidChangeStatus(() => {
-            if (this.interpreters.status === 'idle') {
-                deferred?.resolve();
-                deferred = undefined;
-            } else if (this.interpreters.status === 'refreshing') {
-                if (deferred) {
-                    return;
+
+        this.loadData().then(noop, noop);
+        let combinedProgress: Deferred<void> | undefined = undefined;
+        const updateCombinedStatus = () => {
```

Comment:
Previously we only checked status of one, now we monitor status of 3 items (resulting in more accurate status reports)